maximumLineLength: allow function declarations to exceed limit

clayreimann opened the pull request:
As is this will break existing configurations. It sounded like that was the desired change from the issue, but I can put backwards compatibility back in (with an appropriate warning message).

Fixes #1412

markelog commented on lib/rules/maximum-line-length.js:
```diff
@@ -8,9 +8,11 @@
  *  - `Object`:
  *     - `value`: (required) lines should be at most the number of characters specified
  *     - `tabSize`: (default: `1`) considered the tab character as number of specified spaces
- *     - `allowComments`: (default: `false`) allows comments to break the rule
```
Could you not remove but deprecate them?

markelog commented on lib/rules/maximum-line-length.js:
```diff
@@ -58,9 +63,28 @@ module.exports.prototype = {
+            this._allowFunctionSignature = (exceptions.indexOf('functionSignature') !== -1);
+
+            if (maximumLineLength.hasOwnProperty('allowRegex')) {
+                console.log(DEPRECATED.replace('${option}', this.getOptionName())
```
The thing is, output might not be terminal, but anything user want, this is why we have reporters.

So only mentioning them in the docs as deprecated is fine.

markelog commented on lib/rules/maximum-line-length.js:
```diff
@@ -76,6 +100,7 @@ module.exports.prototype = {
     },
 
     check: function(file, errors) {
+        // console.log('check file: ', file._source);
```
.

markelog commented on lib/rules/maximum-line-length.js:
```diff
@@ -8,9 +8,11 @@
- *     - `allowComments`: (default: `false`) allows comments to break the rule
- *     - `allowUrlComments`: (default: `false`) allows comments with long urls to break the rule
- *     - `allowRegex`: (default: `false`) allows regular expression literals to break the rule
+ *     - `ignore`: (default: `[]`) an array of conditions that will exempt a line
```
Why `ignore`?

clayreimann replied:
`ignore` because I found the wording `allExcept` unclear and not very descriptive. Reverting because the project as a whole appears to prefer `allExcept`.

markelog commented:
btw, github doesn't send notification when you push new changes, so it is preferable to leave a comment when you do

clayreimann commented:
Addressed deprecations in documentation rather than logging errors, also updated some of the test files for presets (since `yandex.js` was the only preset test to address maximum line length).

markelog commented:
Nicely done! Thank you
